libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt

```diff
@@ -1,21 +1,28 @@
 package com.mapbox.navigation.core.routeoptions
 
 import android.location.Location
+import com.mapbox.api.directions.v5.models.Bearing
 import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.geojson.Point
 import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.core.routealternatives.RouteAlternativesController
 import com.mapbox.navigation.core.trip.session.OffRouteObserver
+import com.mapbox.navigation.utils.internal.LoggerProvider
+import kotlin.math.min
+
+private const val DEFAULT_REROUTE_BEARING_TOLERANCE = 90.0
+private const val TAG = "MbxRouteOptionsProvider"
 
 /**
  * Updater is used for *Reroute* and *Route Alternatives* flow.
  *
  * It's used when turn-by-turn navigation goes off-route (see [OffRouteObserver])
  * and when route alternatives (see [RouteAlternativesController]) are requested.
  * For example, this is needed in order to filter the waypoints that have been completed.
- *
- * @see MapboxRouteOptionsUpdater
  */
-fun interface RouteOptionsUpdater {
+class RouteOptionsUpdater {
 
     /**
      * Provides a new [RouteOptions] instance based on the original request options and the current route progress.
```

Comment:
Docs below can be updated, this method does not return `null` but `RouteOptionsResult`.